Emails: Improve UX for Google Workspace account & mail boxes creation

AllTerrainDeveloper opened the pull request:
#### Changes proposed in this Pull Request

This change fixes break-points in Google Workspace comparison page (it is a bit weird plus differs from Professional Email breakpoints), adds a small margin (64px) in the right features pane, and replaces "remove this user" with "remove this mailbox" wording.

#### Testing instructions

1. Go to upgrades -> Emails
2. Visual inspection of the Google Workspace section to add an account/mailbox as per the below gifs

BEFORE | AFTER
------------ | -------------
![non-fixed-breaks](https://user-images.githubusercontent.com/5689927/132539347-bf6a7940-31ef-491b-a143-7369b05600ab.gif) | ![upsell-workspace-google](https://user-images.githubusercontent.com/5689927/133931116-01d7ca57-7c2f-402f-a3ad-becf01588d1f.gif)

It was also needed to update the main header for the comparison page, check its behavior:

BEFORE | AFTER
------------ | -------------
![header-old](https://user-images.githubusercontent.com/5689927/132996170-b52cf4a6-8291-42eb-9835-183e87876967.gif) | ![header-comparison](https://user-images.githubusercontent.com/5689927/133931120-2bbf9b73-8a82-4219-b570-5b2531515a00.gif)

#### Remarks

As part of this change was also needed to update the three icon sizes/behaviors, the Forwarding, Professional Email, and Google Workspace. 

Add mailboxes has been checked too as it might have regression bugs.
![add-mailbox-google](https://user-images.githubusercontent.com/5689927/133931229-cd7413fb-8ed4-47b3-ba56-9867f0976b2c.gif)

Related to {1200182182542585-as-1200879615791996}

AllTerrainDeveloper committed:
Emails: Improve UX for Google Workspace account & mail boxes creation

saygunnyc commented:
Looking pretty cool! @AllTerrainDeveloper !
Sharing some quick notes below.

- Solution logo (Professional Email, Google Workspace, and Email Forwarding) should stay left-aligned on mobile breakpoint too. Sharing designs from Figma file below.

<img width="375" alt="Screen Shot 2021-09-09 at 12 11 55" src="https://user-images.githubusercontent.com/78876323/132658278-2d49f461-d31a-420a-a156-f89f06d7e6e6.png">
______________________
<img width="375" alt="Screen Shot 2021-09-09 at 12 11 37" src="https://user-images.githubusercontent.com/78876323/132658285-5e6d6861-323a-43cf-969d-6474062321a2.png">
______________________
<img width="375" alt="Screen Shot 2021-09-09 at 12 11 45" src="https://user-images.githubusercontent.com/78876323/132658290-a472ffb3-6e17-4e6e-b384-12046f687e34.png">


- All cards should be following the same spacings I shared yesterday for Desktop. I guess that will be handled in different PRs., though sharing here once more for visibility.
> it is a bit weird plus differs from Professional Email breakpoints

<img width="1604" alt="Guides" src="https://user-images.githubusercontent.com/78876323/132659067-48b0d42e-3e1e-461e-9aa7-63aa8fd71c0b.png">


- Not for this PR but wanted to note that we previously have decided to simplify the CTAs and remove plan names from those. "Add" instead of "Add Professional Email" as an example. And moving forward replacing "Add" with "Select" as that's the exact action user is doing at that stage while expanding a card.


- I believe we have decided to use the wording "user" for Google Workspace as that's more than a mailbox, but can't fully recall the decision. cc @daledupreez for further guidance, in terms of consistency.

> replaces "remove this user" with "remove this mailbox" wording.

stephanethomas commented:
Putting aside what Saygun already mentioned this looks good to me.

AllTerrainDeveloper commented:
Thank you for your feedback @saygunnyc! 😍  I will update the PR with the logo alignments. I guess the other things will go to other PR's as this is a task just for Google Workspace 🙂

AllTerrainDeveloper committed:
Emails: Pull request suggestions.

AllTerrainDeveloper commented:
As part of the new changes proposed by @saygunnyc it was needed to modify another component that are involved in another area of calypso. I've noted two of them, but there is a third one that I cannot locate through the UI, and it's one where we are prompted with the following question: **Want your site to look like the demo? Use Theme Setup to automatically apply the demo site's settings to your site.**. Do anyone knows how to test this?

stephanethomas commented on client/components/action-panel/style.scss:
```diff
@@ -1,8 +1,15 @@
 .action-panel {
 	padding-bottom: 16px;
+	flex-direction: column-reverse;
+	&.is-primary {
```
We could add a blank line above to better separate that inner block;

stephanethomas reviewed (commented):
I see that we are now changing styles from the Action Panel and Promo Card components. This makes me nervous 🙂.

Those components are used in many different places, so the chance that we break another part of WordPress.com is something we should take into account. First, do we really need to update those shared components? As far I can see, we just removed a breakpoint so maybe we could keep it even if that's suboptimal. If we still want to get rid of them, then could we split those changes into several pull requests? That way, it will be easier to identify and revert those changes if something doesn't look right on WordPress.com.

AllTerrainDeveloper commented:
> I see that we are now changing styles from the Action Panel and Promo Card components. This makes me nervous 🙂.
> 
> Those components are used in many different places, so the chance that we break another part of WordPress.com is something we should take into account. First, do we really need to update those shared components? As far I can see, we just removed a breakpoint so maybe we could keep it even if that's suboptimal. If we still want to get rid of them, then could we split those changes into several pull requests? That way, it will be easier to identify and revert those changes if something doesn't look right on WordPress.com.

@stephanethomas I'll do my best to avoid touching unnecessary css code :)

AllTerrainDeveloper committed:
Emails: Redo css logic to be part only of emails-comparisson css stylesheet

AllTerrainDeveloper commented:
@stephanethomas I've managed to undo the changes in _action-panel_ and _promo-card_ changes there shouldn't be now any regression, those changes are only applied as **emails-providers-comparison/style.css** now.

I have not updated screenshots as they remain the same as with the previous commit.

daledupreez commented on client/my-sites/email/email-providers-comparison/style.scss:
```diff
@@ -1,6 +1,44 @@
 @import '@wordpress/base-styles/breakpoints';
 @import '@wordpress/base-styles/mixins';
 
+.action-panel {
+	flex-direction: row;
+
+	&.is-primary {
+		flex-direction: column;
+		@include break-mobile {
+			flex-direction: row;
+		}
+	}
+}
+
+.action-panel__figure {
+	float: right;
+	max-width: 208px;
+
+	&.is-inline-body-text {
+		margin-top: 40px;
+	}
+
+	&.align-left {
+		float: left;
+		margin-left: 0;
+		margin-right: 24px;
+		width: unset;
+	}
+
+	&.align-right {
+		margin: 0 0 24px 24px;
+	}
+
+	&.align-left {
+		float: left;
+		margin-left: 0;
+		margin-right: 24px;
+	}
+}
```
@AllTerrainDeveloper, please qualify these styles. Right now, the behaviour will depend on the load order for the relevant CSS files, when we only want these styles to apply to these elements that are within the email comparison component.

Can you make these beneath one of the wrapping CSS classes for the component to make it explicit that these styles should only show up for this page?

AllTerrainDeveloper replied:
Good point. I've updated the CSS, and I have not updated screenshots as they are behaving the same as with the previous commit.

AllTerrainDeveloper committed:
Emails: Better CSS styling

AllTerrainDeveloper commented on client/my-sites/email/email-providers-comparison/style.scss:
```diff
@@ -194,7 +234,7 @@
 
 			@include breakpoint-deprecated( '>1040px' ) {
```
I've missed this breakpoint conversion. I will take care of it on the next 20% allocated time for improvements

daledupreez reviewed (changes requested):
This is looking good for the email comparison page, but I discovered that the changes introduce problems for the Add New Mailboxes page for Google:

<img width="677" alt="Screenshot 2021-09-15 at 13 07 28" src="https://user-images.githubusercontent.com/3376401/133423139-f64c0bfe-f6c9-4b16-86c5-dea259430494.png">

I also confirmed that this problem occurs for the email upsell in the domain purchase flow (`domains/add/:new_domain/email/:site_slug`)

If we don't want this kind of visual regression/bug, we also need to use the new breakpoints in `client/my-sites/email/gsuite-add-users/style.scss`, which is currently flipping `flex-direction` from `row` to `column` at a different breakpoint.

AllTerrainDeveloper committed:
Emails: Fix CSS breaks for mobile & desktop on emails comparison, add mailbox and upsell

AllTerrainDeveloper commented:
I have updated all the mixins, it should be working in emails-comparison page, upsell and add mailbox. I am updating screenshots soon.

stephanethomas commented on client/my-sites/email/email-providers-comparison/index.jsx:
```diff
@@ -459,6 +459,8 @@ class EmailProvidersComparison extends React.Component {
 				className="email-providers-comparison__providers-wordpress-com-email"
 				icon="my-sites"
+				height="40"
+				width="40"
```
Shouldn't we define those attributes via CSS?

stephanethomas commented on client/my-sites/email/email-providers-comparison/style.scss:
```diff
@@ -1,6 +1,17 @@
 @import '@wordpress/base-styles/mixins';
 
+.email-providers-comparison__action-panel {
+	flex-direction: row;
```
We don't need that line as `row` is the [default value](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction) for `flex-direction`.

stephanethomas commented on client/my-sites/email/email-providers-comparison/style.scss:
```diff
@@ -60,12 +71,30 @@
 }
 
 .email-providers-comparison__provider-card.promo-card {
+	flex-direction: row;
```
Same remark than above.

AllTerrainDeveloper committed:
Emails: Amend Stéphane feedback

AllTerrainDeveloper committed:
Emails: Olaseni spotted an issue when the device is smaller than 481px, fix for that.

olaseni reviewed (approved):
Thanks for pairing with me over this! This looks good! We can iterate in subsequent PRs.
